### ISSUE TITLE
system hang after a while

### ISSUE THREAD

**Issue by nicolasdb** [nicolasdb/DoorLock-M5stampC3U#1]
shell messages:
```
[BOOT] Importing main module...
[BOOT] Main module imported, starting run()...
[MAIN] Starting...
[DOOR] Initializing door control...
[DOOR] Closing door...
[DOOR] Door initialized to closed state
[MQTT] Connected to Adafruit IO MQTT Broker!
[MQTT] Subscribed to command feed: nicolasof/feeds/doorlock-command
[MQTT] Published door status: 0 to nicolasof/feeds/doorlock-status
[MAIN] Setup complete, entering main loop...
[DOOR] Opening door...
[MQTT] Published door status: 1 to nicolasof/feeds/doorlock-status
[DOOR] Door timeout reached, closing door
[DOOR] Closing door...
[MQTT] Published door status: 0 to nicolasof/feeds/doorlock-status
[MQTT] Received command to open door
[DOOR] Opening door...
[MQTT] Published door status: 1 to nicolasof/feeds/doorlock-status
[MQTT] Door already in requested state: 1
[DOOR] Door timeout reached, closing door
[DOOR] Closing door...
[MQTT] Published door status: 0 to nicolasof/feeds/doorlock-status
[MAIN] Error in main loop: -1
[MAIN] Error in main loop: -1
[MAIN] Error in main loop: -1
[MAIN] Error in main loop: -1
```

a reboot solve the problem but we run on it again.

**Comment by nicolasdb** [comment]
> I see the issue now. The MQTT client is being created without a keepalive parameter, which defaults to 0 in umqtt.simple. This means:
> - No ping messages are being sent to keep the connection alive
> - Most MQTT brokers (including Adafruit IO) will disconnect clients after a period of inactivity (typically 5 minutes)
> - When the connection drops, client.check_msg() returns -1

Let's modify the MQTT client setup to include a proper keepalive value and add reconnection logic.

**Comment by nicolasdb** [comment]
solved with commit #f819dfb
